### DISCUSSION TITLE
Processing job input and output cannot be the same

### DISCUSSION THREAD

**Discussion by RoelantStegmann** [aws/sagemaker-python-sdk#2398] [Help]
**Describe the bug**
Sagemaker requires the input and output to be different folders (for no apparent reason) 

**To reproduce**
```
proc = Processor(
   ....
)
proc.run(
    inputs=[
         ProcessingInput(
            source='s3://processing-data/test/',
            destination='/opt/ml/test/'
        ),
    ],
    outputs=[
        ProcessingOutput(
            source=f'/opt/ml/processing/test/',
            destination=f's3://processing-data/test',
            s3_upload_mode="Continuous"
        ),
    ],
    arguments=[
    ],
)
```

**Expected behavior**

You would expect this to be allowed, but it runs into validation errors that apparently compares the input and output paths.

**Comment by chuyang-deng** [comment]
Hi @RoelantStegmann, are you using your own processing container? SageMaker will go to specific locations in the container for input data and output data, here's how SageMaker handle input/output. https://docs.aws.amazon.com/sagemaker/latest/dg/build-your-own-processing-container.html#byoc-input-and-output

Also could you provide stacktrace of the validation error so that we could see if this validation is necessary and can be re-evaluated at Python SDK level?
